lib/NativeMap.ts

```diff
@@ -26,20 +25,26 @@ const mapOptions = (details: EventMapDetails) => {
   }
 }
 
-export const openMap = (details: EventMapDetails) => {
-  showLocation(mapOptions(details))
+export const openInMaps = (details: NativeEventMapDetails) => {
+  showLocation(nativeMapOptions(details))
 }
 
-export const withDirections = 
-  async (userLocation: UserLocationTrackingUpdate | undefined, details: EventMapDetails) => {
-  const options = mapOptions(details)
-  const userCoordinates =
+export const openMapDirections = async (
+  userLocation: UserLocationTrackingUpdate | undefined,
+  details: NativeEventMapDetails
+) => {
+  const options = nativeMapOptions(details)
+  let userCoordinates =
     userLocation && userLocation.status === "success"
       ? userLocation.location.coordinates
       : undefined
 
   if (!userCoordinates) {
-    await Location.requestForegroundPermissionsAsync()
+    await requestLocationPermissions()
+    userCoordinates =
+      userLocation && userLocation.status === "success"
+        ? userLocation.location.coordinates
+        : undefined
```

Comment:
I don't think we need this check. It should be possible to just open maps with destination coordinates and have it figure out the rest.